Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenStick.cs

```diff
@@ -227,11 +229,20 @@ private void EndInteraction()
 
         private void OnPointerDown(InputAction.CallbackContext ctx)
         {
+            if (m_IsIsolationActive) { return; }
             Debug.Assert(EventSystem.current != null);
 
             var screenPosition = Vector2.zero;
-            if (ctx.control?.device is Pointer pointer)
+            TouchControl touchControl = null;
+            if (ctx.control?.parent is TouchControl touch)
```

Comment:
Unfortunately this is the design, generally this is a perfect fit for a device-agnostic pointer interface (just a side note)